core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -29,7 +29,7 @@
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
 
-class BaseFileScanTask implements FileScanTask {
+public class BaseFileScanTask implements FileScanTask {
```

Comment:
Needs to be public so we can build these in PlanScanAction

Reply:
I think maybe we should have a factory method somewhere in the public API instead? We don't necessarily want to expose everything here.

Reply:
Sounds reasonable, let me see if I can change that